Decode/EventHeaderEnumerator.cs

```diff
@@ -1545,9 +1537,9 @@ public void AppendJsonEventIdentityTo(ReadOnlySpan<byte> eventDataSpan, StringBu
                     case EventHeaderEnumeratorState.Value:
 
                         itemInfo = GetItemInfo(eventDataSpan);
-                        if (wantName && !itemInfo.Value.Type.IsArrayOrElement)
+                        if (wantName && !itemInfo.Value.Metadata.IsElement)
```

Comment:
Note: Now that the metadata (was type) class has an IsElement property, we can say what we mean here.